build/commands/lib/test.js

```diff
@@ -70,6 +70,10 @@ const buildTests = async (suite, buildConfig = config.defaultBuildConfig, option
   ]
   if (testSuites.includes(suite)) {
     config.buildTargets = ['brave/test:' + suite]
+    if (suite === 'brave_unit_tests' && config.targetOS === 'android') {
+      // We need to build the APK for the java tests as well.
+      config.buildTargets.push('brave/test:brave_public_test_apk')
```

Comment:
this is not the correct place to add this, if it's a dep for brave_unit_tests on android then it should be added in gn

Reply:
cc: @samartnik

Reply:
`brave_unit_tests` doesn't require it as a dep per se. It used to be added as a dep just in oder to be built. `brave_public_test_apk` should probably be a separate suite as those 2 are technically unrelated.

Reply:
Created a follow up issue to make it separate suite as it also require changes to CI to run this suite for android builds https://github.com/brave/brave-browser/issues/40885